### PR TITLE
Handle exit code 100 from 'yum check-update'

### LINKED ISSUE #635: nodeup.UpdatePackages fails if 'yum check-update' has updates
An instance running a distro in the RHEL family with outdated yum packages fails to come up because the UpdatePackages task fails:

```
Oct 11 17:48:13 localhost cloud-init: W1011 17:48:13.988202    5124 executor.go:84] error running task "UpdatePackages": error update packages: exit status 10
0: Loaded plugins: fastestmirror
```

I believe this is due to `yum check-update` returning an exit value of `100` if there are package updates. The issue seems to be the conditional here: https://github.com/kubernetes/kops/blob/4bca0a37d17ef9c91baa13ab91af56d38ff0990f/upup/pkg/fi/nodeup/nodetasks/update_packages.go#L63

### PR DESCRIPTION
**Pull request by austinmoore-**
Fixes #635

One thing to note with this PR is that it imports `syscall`. In the docs it says:
>The primary use of syscall is inside other packages that provide a more portable interface to the system, such as "os", "time" and "net". Use those packages rather than this one if you can.

An alternative way to fix this that is not dependent on `syscall` is to do something like:

```
if state := cmd.ProcessState; err != nil && state.String() != "exit status 100" {
        return fmt.Errorf("error update packages: %v: %s", err, string(output))
}
```

Despite the message in the docs, the usage of `syscall` seems appropriate to me. The risk is low, because the usage is limited, and it seems better to compare the exit code against an int, rather than a string. If someone feels we shouldn't be using `syscall` feel free to say so.

### CODE REVIEW

**Comment by austinmoore-** [comment]
I signed it!

**Comment by chrislovecnm** [comment]
@austinmoore- one more SLA for yah as well.  I believe they are just transitioning to the linuxfoundation one soon.

**Comment by austinmoore-** [comment]
I signed it!

**Comment by mrichmon** [comment]
I've bubbled this CLA issue up to the CNCF committee via their Slack channel. The CNCF is the committee in the Linux Foundation that Kubernetes was transferred over to earlier this year.

File: upup/pkg/fi/nodeup/nodetasks/update_packages.go [hunk 1]
```diff
@@ -60,7 +61,8 @@ func (_ *UpdatePackages) RenderLocal(t *local.LocalTarget, a, e, changes *Update
 	glog.Infof("running command %s", args)
 	cmd := exec.Command(args[0], args[1:]...)
 	output, err := cmd.CombinedOutput()
-	if err != nil {
+	// 'yum check-update' exits with 100 if it finds updates; treat it like a success
+	if exitCode := cmd.ProcessState.Sys().(syscall.WaitStatus).ExitStatus(); err != nil && exitCode != 100 {
```
Comment (justinsb) [inline upup/pkg/fi/nodeup/nodetasks/update_packages.go]: I do worry that there may be some errors where cmd.ProcessState is nil (if we fail to even find yum, for example, though admittedly that seems unlikely).  But this LGTM anyway.

**Comment by justinsb** [comment]
Thanks @austinmoore-

**Pull request merged**